src/inmanta/export.py

```diff
@@ -370,7 +373,7 @@ def run(
             # then process the configuration model to submit it to the mgmt server
             # This is the actuel export : convert entities to resources.
             self._load_resources(types)
-
+            resource_sets_to_remove += self._empty_resource_sets
```

Comment:
That was what I wanted to do. I thought it wouldn't cause any issue as we potentially already modify the input object once in this function (I think):
```
if resource_sets_to_remove is None:
            resource_sets_to_remove = []
```

Reply:
That's not accurate, that's just binding the name to a new value (which can sometimes hinder readability as well but in this case should be acceptable).
```python
>>> def f(l: list[int]) -> list[int]:
...     l = [1000, 2000]
...     return l
... 
>>> l = [1, 2]
>>> f(l)
[1000, 2000]
>>> l
[1, 2]
```

See the difference?

Reply:
I see the difference but then there is something i don't understand: I dont think I modify the input object as what is happening is this:
```
>>> def f(l: list[int]) -> list[int]:
...     l = [5, 6]
...     l += [1000, 200]
...     return l
... 
>>> l = [1,2,3]
>>> l
[1, 2, 3]
>>> f(l)
[5, 6, 1000, 200]
>>> l
[1, 2, 3]
>>> 
```
Or am I still missing something? :p

Reply:
O no I see, I understand :p the first condition is not always executed :p

Reply:
You're partially right there, I hadn't considered that. But that's only true if `resource_sets_to_remove is None`. If the caller passes in a list, it is still modified.
```python
>>> from typing import Optional
>>> def f(l: Optional[list[int]] = None) -> list[int]:
...     if l is None:
...             l = [5, 6]
...     l += [1000, 2000]
...     return l
... 
>>> l = [1, 2, 3]
>>> f()
[5, 6, 1000, 2000]
>>> l
[1, 2, 3]
>>> f(l)
[1, 2, 3, 1000, 2000]
>>> l
[1, 2, 3, 1000, 2000]
```

Reply:
Maybe I should have included a solution as well: there's a bunch of things you can do, the one closest to what you currently have being to replace `l += l2` with `l = l + l2`. There's still the reassignment I'm not a huge fan of, but at least there are no side effects outside of the scope of this method.
```python
>>> def f(l: Optional[list[int]] = None) -> list[int]:
...     if l is None:
...             l = [5, 6]
...     l = l + [1000, 2000]
...     return l
... 
>>> l = [1, 2, 3]
>>> f()
[5, 6, 1000, 2000]
>>> l
[1, 2, 3]
>>> f(l)
[1, 2, 3, 1000, 2000]
>>> l
[1, 2, 3]
```

Reply:
> O no I see, I understand :p the first condition is not always executed :p

Indeed. Race condition in our comment stream here. My comments were in response to the one before this one.

Reply:
> the one closest to what you currently have being to replace `l += l2` with `l = l + l2`.

And perhaps change the typing to use `abc.Sequence` instead of `list`, so mypy can catch things like this.

Reply:
It is, I actually slightly prefer what you did, it makes it more explicit that we'll be adding to it later on. If you prefer you can safely go back to `+=` now, since it's a local object now.